### PR TITLE
add condition in sql/mongo tracker store in additional_events when retrieve_events_from_previous_conversation_sessions is true

### PR DESCRIPTION
**Pull request by melindaloubser1**
**Proposed changes**:
- closes https://github.com/RasaHQ/rasa/issues/8238
- closes https://github.com/RasaHQ/rasa/issues/8237
- fix calculation of `_additional_events` when `retrieve_events_from_previous_conversation_sessions=true`

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/tracker_store.py [hunk 1]
```diff
@@ -1109,9 +1113,11 @@ def _additional_events(
     ) -> Iterator:
         """Return events from the tracker which aren't currently stored."""
 
+
```
Comment (melindaloubser1) [inline rasa/core/tracker_store.py:1116]:
```suggestion
```

File: rasa/core/tracker_store.py [hunk 2]
```diff
@@ -1109,9 +1113,11 @@ def _additional_events(
...
         number_of_events_since_last_session = self._event_query(
-            session, tracker.sender_id, fetch_events_from_all_sessions=False
+            session, tracker.sender_id, fetch_events_from_all_sessions=self.retrieve_events_from_previous_conversation_sessions
         ).count()
+
```
Comment (melindaloubser1) [inline rasa/core/tracker_store.py:1120]:
```suggestion
```

File: rasa/core/tracker_store.py [hunk 3]
```diff
@@ -611,9 +611,13 @@ def _additional_events(self, tracker: DialogueStateTracker) -> Iterator:
 
         stored = self.conversations.find_one({"sender_id": tracker.sender_id}) or {}
         all_events = self._events_from_serialized_tracker(stored)
-        number_events_since_last_session = len(
-            self._events_since_last_session_start(all_events)
-        )
+
```
Comment (melindaloubser1) [inline rasa/core/tracker_store.py:614]:
```suggestion
```

File: changelog/8258.bugfix.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Fixed the bug that events from previous conversation sessions would be re-saved in the SQL or Mongo tracker stores when `retrieve_events_from_previous_conversation_sessions` was true.
```
Comment (wochinge) [inline changelog/8258.bugfix.md:1]: how about adding a link to the tracker store docs?

**Review by wochinge — COMMENTED** [review]
Change looks good overall - could you please add a test to avoid regression?

File: changelog/8258.bugfix.md (hunk 4 above)
Comment (melindaloubser1) [reply changelog/8258.bugfix.md:1]:
```suggestion
Fixed the bug that events from previous conversation sessions would be re-saved in the [`SQLTrackerStore`](https://rasa.com/docs/rasa/tracker-stores#sqltrackerstore) or [`MongoTrackerStore`](https://rasa.com/docs/rasa/tracker-stores#mongotrackerstore) when `retrieve_events_from_previous_conversation_sessions` was true. 
```

Comment (wochinge) [reply changelog/8258.bugfix.md:1]: let's use the `.mdx` references as in the other changelogs instead of the absolute paths.

Comment (melindaloubser1) [reply changelog/8258.bugfix.md:1]:
```suggestion
Fixed the bug that events from previous conversation sessions would be re-saved in the [`SQLTrackerStore`](./tracker-stores.mdx#sqltrackerstore) or [`MongoTrackerStore`](./tracker-stores.mdx#mongotrackerstore) when `retrieve_events_from_previous_conversation_sessions` was true. 
```

**Comment by melindaloubser1** [comment]
I parametrized the existing tests to confirm same behaviour - is there another side that should be tested?

**Comment by melindaloubser1** [comment]
`tests/core/test_evaluation.py::test_e2e_with_entity_evaluation` is failing although it passes locally; can it be because the [timeout for the test (240)](https://github.com/RasaHQ/rasa/pull/8258/checks?check_run_id=2173751801#step:12:70)  is longer than the [timeout for the pytest session (120s)](https://github.com/RasaHQ/rasa/pull/8258/checks?check_run_id=2173751801#step:12:26)?

File: tests/core/conftest.py [hunk 5]
```diff
@@ -84,11 +84,18 @@ def __init__(self, *args, **kwargs):
 class MockedMongoTrackerStore(MongoTrackerStore):
     """In-memory mocked version of `MongoTrackerStore`."""
 
-    def __init__(self, _domain: Domain):
+    def __init__(
+        self,
+        _domain: Domain,
+        retrieve_events_from_previous_conversation_sessions: bool = False,
+    ):
```
Comment (wochinge) [inline tests/core/conftest.py:91]:
```suggestion
    ) -> None:
```

**Review by wochinge — APPROVED** [review]
Thanks for the fix! The `parametrize` updates to the tests are amazing 💯 🥇

File: changelog/8258.bugfix.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Fixed the bug that events from previous conversation sessions would be re-saved in the [`SQLTrackerStore`](https://rasa.com/docs/rasa/tracker-stores#sqltrackerstore) or [`MongoTrackerStore`](https://rasa.com/docs/rasa/tracker-stores#mongotrackerstore) when `retrieve_events_from_previous_conversation_sessions` was true.
```
[the remark this answers is not in the document]
Comment (melindaloubser1) [reply changelog/8258.bugfix.md]: grr these musta changed back during the rebase oops

**Pull request merged**